graph.go

```diff
@@ -303,42 +312,94 @@ func treePath(path string) string {
 	return "./" + path
 }
 
-func doGraph(pkgs []*build.Package) (*gog.Output, error) {
-	build.Default = *loaderConfig.Build
+func doGraph(pkg *build.Package) (*gog.Output, error) {
+	var allGoFiles []string
+	allGoFiles = append(allGoFiles, pkg.GoFiles...)
+	allGoFiles = append(allGoFiles, pkg.CgoFiles...)
+	allGoFiles = append(allGoFiles, pkg.TestGoFiles...)
+	if len(allGoFiles) == 0 {
+		return &gog.Output{}, nil
+	}
 
-	for _, pkg := range pkgs {
-		var allGoFiles []string
-		allGoFiles = append(allGoFiles, pkg.GoFiles...)
-		allGoFiles = append(allGoFiles, pkg.CgoFiles...)
-		allGoFiles = append(allGoFiles, pkg.TestGoFiles...)
-		for i, f := range allGoFiles {
-			allGoFiles[i] = filepath.Join(cwd, pkg.Dir, f)
+	fset := token.NewFileSet()
+	var files []*ast.File
+	for _, name := range allGoFiles {
+		file, err := parser.ParseFile(fset, filepath.Join(pkg.Dir, name), nil, parser.ParseComments)
+		if err != nil {
+			return nil, err
 		}
-		loaderConfig.CreateFromFilenames(pkg.ImportPath, allGoFiles...)
+		files = append(files, file)
 	}
 
-	prog, err := loaderConfig.Load()
+	typesConfig := &types.Config{
+		Importer: &buildContextImporter{
+			context:  &buildContext,
+			srcDir:   pkg.Dir,
+			packages: make(map[string]*types.Package),
+		},
+		FakeImportC: true,
+		Error: func(err error) {
+			// errors are ignored, use best-effort type checking output
+		},
+	}
+	typesInfo := &types.Info{
+		Defs:       make(map[*ast.Ident]types.Object),
+		Uses:       make(map[*ast.Ident]types.Object),
+		Implicits:  make(map[ast.Node]types.Object),
+		Selections: make(map[*ast.SelectorExpr]*types.Selection),
+		Scopes:     make(map[ast.Node]*types.Scope),
+	}
+	typesPkg, err := typesConfig.Check(pkg.ImportPath, fset, files, typesInfo)
+	if err != nil {
+		log.Println("type checker error:", err) // see comment above
+	}
+
+	return gog.Graph(fset, files, typesPkg, typesInfo, true), nil
+}
```

Comment:
I did not look and verify the exact code details, and I agree it's best to do that by testing the final version on real code and making sure it works on popular libraries and the Go repo, etc.

However, at a high level, this looks like it's using the right tools (Go packages) for the job, and the approach is very reasonable and similar to what I've done/seen in the past. So LGTM.